### ISSUE TITLE
Documents

### ISSUE THREAD

**Issue by vultuk** [ChiefSupportSystems/ChiefSupportSystems#10]
It must be possible for users to create and upload documents themselves with no interaction from tech support.

**Comment by beeena** [comment]
Hi @vultuk !

I'm almost done but the template file upload api is not available yet.

![image](https://user-images.githubusercontent.com/38552626/84580422-459e9b00-adf0-11ea-9f28-44cf6814c447.png)

**Comment by beeena** [comment]
@vultuk I've a question, hope you don't mind :) 

Do we have to upload the template when the `save` button is clicked or we have to upload when the file is browsed or dropped to the template?

**Comment by vultuk** [comment]
@beeena The file can be uploaded when selected, or on pressing save. I’m happy with either method. If you choose to upload the file when selected please make sure the save button is disabled until the file is uploaded.

Hope that all makes sense. 🙂

**Comment by beeena** [comment]
@vultuk Yes makes sense to me but the API isn't working for me. Can you please look into the API `/templates/upload`

**Comment by beeena** [comment]
@vultuk Do we have any documentation for the back-end API's?

**Comment by beeena** [comment]
Hi @vultuk,

I hope you are doing great. I've created an API for uploading the template but I'm having issue while getting the template url from s3 storage. I think it would be completed today. I'll update you and push the code once I fix the issues.

Thanks

**Comment by beeena** [comment]
Hi @vultuk,

I'm done with the task and I'm looking forward to your feedback 🙂

**Comment by beeena** [comment]
I've push the code to [Github](https://github.com/tech3k/chief-portal/compare/branch-template-upload?expand=1)
